packages/state-transition/src/cache/epochTransitionCache.ts

```diff
@@ -360,6 +366,30 @@ export function beforeProcessEpoch(
     }
   }
 
+  // Trigger async build of shuffling for epoch after next (nextShuffling post epoch transition)
+  const epochAfterUpcoming = state.epochCtx.nextEpoch + 1;
+  // cannot call calculateShufflingDecisionRoot here because spec prevent getting current slot
+  // as a decision block.  we are part way through the transition though and this was added in
+  // process slot beforeProcessEpoch happens so it available and valid
+  const nextShufflingDecisionRoot = toRootHex(state.blockRoots.get(state.slot % SLOTS_PER_HISTORICAL_ROOT));
+  const _nextShufflingActiveIndices = new Array<number>(nextEpochShufflingActiveIndicesLength);
+  if (nextEpochShufflingActiveIndicesLength > nextEpochShufflingActiveValidatorIndices.length) {
+    throw new Error(
+      `Invalid activeValidatorCount: ${nextEpochShufflingActiveIndicesLength} > ${nextEpochShufflingActiveValidatorIndices.length}`
+    );
+  }
+  // only the first `activeValidatorCount` elements are copied to `activeIndices`
+  for (let i = 0; i < nextEpochShufflingActiveIndicesLength; i++) {
+    _nextShufflingActiveIndices[i] = nextEpochShufflingActiveValidatorIndices[i];
+  }
+  const nextShufflingActiveIndices = new Uint32Array(_nextShufflingActiveIndices);
```

Comment:
we do not need `_nextShufflingActiveIndices`, only need to allocate `nextShufflingActiveIndices` and assign each item in the above for loop

Reply:
Ok.  Had it like that because I ran into issues mixing Uint32Array values and number before but forget the context.  Made the change and tests seem to pass so all is gravy 😄